augur/augur_client.py

```diff
@@ -44,76 +44,105 @@ def __init__(self, hostname, port):
-      'creation_fee': Decimal,
-      'settlement_fee': Decimal,
-      'reporting_fee_rate': Decimal,
-      'market_creator_fee_rate': Decimal,
-      'market_creator_fees_balance': Decimal,
-      'market_creator_mailbox': str,
-      'market_creator_mailbox_owner': str,
-      'initial_report_size': Decimal,
-      'category': str,
-      'tags': list,
-      'volume': Decimal,
-      'open_interest': Decimal,
-      'outstanding_shares': Decimal,
-      'reporting_state': ReportingState,
-      'end_time': datetime,
-      'finalization_time': datetime,
-      'last_trade_time': datetime,
-      'description': str,
-      'details': str,
-      'scalar_denomination': str,
-      'designated_reporter': str,
-      'designated_report_stake': Decimal,
-      'resolution_source': str,
-      'num_ticks': Decimal,
-      'tick_size': Decimal,
-      'consensus': NormalizedPayout,
-      'outcomes': OutcomeInfo,
-    }
-    for key, value in response_data.items():
-      new_key = self._to_snake_case(key)
-      to_cast_type = casting_map.get(new_key, None)
-      if not value:
-        if to_cast_type is list:
-          new_value = [x for x in value if x is not None]
-        elif to_cast_type is str:
-          new_value = '' if not value else value
-        else:
-          new_value = None
-      elif to_cast_type is None:
-        new_value = value
-      elif to_cast_type is MarketInfo.Type or to_cast_type is ReportingState:
-        new_value = to_cast_type[value.upper()]
-      elif to_cast_type is NormalizedPayout:
-        if not value['payout']:
-          value['payout'] = list()
-        new_value = to_cast_type(value['isInvalid'], value['payout'])
-      elif to_cast_type is OutcomeInfo:
-        for x in value:
-          if not x['description']:
-            x['description'] = ''
-        new_value = [OutcomeInfo(**properties) for properties in value]
-      elif to_cast_type is datetime:
-        # Augur can store timestamps in microseconds which will raise an
-        # OSError for datetime.fromtimestamp, which requires the timestamp
-        # be in seconds.
-        new_value = to_cast_type.fromtimestamp(int(str(value)[:10]))
-      else:
-          new_value = to_cast_type(value)
-      market_info_dict[new_key] = new_value
-    return MarketInfo(**market_info_dict)
+    # Ensure timestamps are in seconds.
+    response_data['creationTime'] = (
+      self._ensure_unix_timestamp(response_data['creationTime']))
+    response_data['endTime'] = (
+      self._ensure_unix_timestamp(response_data['endTime']))
+    # finalization_time and last_trade_time will default to None
+    if response_data['finalizationTime']:
+      response_data['finalizationTime'] = (
+        self._ensure_unix_timestamp(response_data['finalizationTime']))
+    if response_data['lastTradeTime']:
+      response_data['lastTradeTime'] = (
+        self._ensure_unix_timestamp(response_data['lastTradeTime']))
+    # Ensure decimals default to None if needed.
```

Comment:
Can you explain what you mean by defaulting to ```None``` if needed? ```None``` is not a valid ```Decimal``` and will raise an exception.

Reply:
It looks like the default for `initialReportSize`, `finalizeBlockNumber`, `lastTradeBlockNumber` is already `None`, so lines 60-68 isn't needed. I will need to ensure that if we do get None that it doesn't get cast to a `Decimal`.